### PR TITLE
test(switch): refactor for cypress tests

### PR DESCRIPTION
**Pull request by stephenogorman**
### Proposed behaviour

Refactor Cypress tests for the `Switch` component to use new cypress-component-react framework for testing.
Add `switch.test.js`. 

### Current behaviour

Current process is the regular Cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there is newly added test.file
- [x] Check if the `switch.test.js` file passed
- [x] Check all tests were moved/refactored from .feature file -> .test.js file
- [x] Check all unnecessary files were deleted, including .json file
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed

### CODE REVIEW

File: src/components/switch/switch.test.js [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+    );
+
+    it("should render Switch component as a required field", () => {
+      CypressMountWithProviders(<SwitchComponent required />);
+
+      verifyRequiredAsteriskForLabel();
+    });
+
+    it.each(["bottom", "left", "right", "top"])(
+      "should render CheckboxComponent component with tooltip positioned to the %s",
```
Comment (DlgSHi) [inline src/components/switch/switch.test.js:441]: checkboxComponent?

File: src/components/switch/switch.test.js [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+    let callback;
+
+    beforeEach(() => {
+      callback = cy.stub();
+    });
+
+    it("should call onChange callback when a click event is triggered", () => {
+      CypressMountWithProviders(<SwitchComponent onChange={callback} />);
+
+      switchInput()
```
Comment (DlgSHi) [inline src/components/switch/switch.test.js:480]: I think we can add a TC with toggling `Switch` with `Space` keyboard

**Review by DlgSHi — CHANGES_REQUESTED** [review]
One TC with `defaultChecked` is missing

**Review by DlgSHi — DISMISSED** [review]
LGTM only squash the commits :)

**Review by DianeGit22 — APPROVED** [review]

**Pull request merged**